packages/default-theme/components/checkout/steps/PaymentStep.vue

```diff
@@ -135,52 +48,35 @@
   </div>
 </template>
 <script>
-import {
-  SfHeading,
-  SfInput,
-  SfButton,
-  SfRadio,
-  SfImage,
-  SfCheckbox,
-} from '@storefront-ui/vue'
+import { SfHeading, SfButton, SfRadio } from '@storefront-ui/vue'
 import BillingAddressGuestForm from '@shopware-pwa/default-theme/components/checkout/steps/guest/BillingAddressGuestForm'
 import BillingAddressUserForm from '@shopware-pwa/default-theme/components/checkout/steps/user/BillingAddressUserForm'
-import { useCheckout } from '@shopware-pwa/composables'
+import { useCheckout, useSessionContext } from '@shopware-pwa/composables'
+import { onMounted, computed } from '@vue/composition-api'
 
 export default {
   name: 'PaymentStep',
   components: {
     SfHeading,
-    SfInput,
     SfButton,
     SfRadio,
-    SfImage,
-    SfCheckbox,
     BillingAddressGuestForm,
     BillingAddressUserForm,
   },
-  data() {
-    return {
-      paymentMethod: '',
-      paymentMethods: [], // useCheckout
-      cardNumber: '',
-      cardHolder: '',
-      cardMonth: '',
-      cardYear: '',
-      cardCVC: '',
-      cardKeep: false,
-      countries: [], // useCountries
-    }
-  },
   setup() {
-    const { isGuestOrder } = useCheckout()
+    const { isGuestOrder, getPaymentMethods, paymentMethods } = useCheckout()
+    const { paymentMethod, setPaymentMethod } = useSessionContext()
 
-    return { isGuestOrder }
-  },
-  computed: {
-    isCreditCard() {
-      return ['debit', 'mastercard', 'electron'].includes(this.paymentMethod)
-    },
+    const activePaymentMethod = computed({
+      get: () => paymentMethod.value && paymentMethod.value.id,
+      set: async (id) => await setPaymentMethod({ id }),
+    })
+
+    onMounted(async () => {
```

Comment:
using onMounted in component is a nice way to avoid exposing onMounted callback in composable to get 100% coverage when onMounted is being used in composable ;) nice :+1: